paddle/framework/operator.h

```diff
@@ -457,6 +457,16 @@ class RuntimeInferShapeContext : public InferShapeContext {
     return true;
   }
 
+  const Variable* InputVar(const std::string& name) const {
```

Comment:
why should add two special interface

Reply:
@jacquesqiao  Since we will set var type in compile-time. And our programDesc will pass to Executor to create variable. So we should set variable in Executor. Like `GetMutable<Tensor>/GetMutable<SelectedRows>/GetMutable<LoDTensor>`.
At run-time, we can just get tensor to set dims and get dims. 
Currently, we do not implement variable type inference in Executor. And variable type is set to LoDTensor by default at run-time infershape stage.
I add these two interfaces to get a variable at run-time infershape stage and set variable type manaully. This will be fixed if our whole compile-time framework is done.